src/Components/Web/src/WebRenderer.cs

```diff
@@ -140,16 +133,20 @@ public Task DispatchEventAsync(JsonElement eventDescriptor, JsonElement eventArg
                 webEventData.EventArgs);
         }
 
+        [JSInvokable]
+        public void RemoveRootComponent(int componentId)
+            => _renderer.RemoveRootComponent(componentId);
+
         [JSInvokable] // Linker preserves this if you call RootComponents.Add
-        public int AddRootComponent(string identifier, string domElementSelector)
+        public int AddJSRootComponent(string identifier, string domElementSelector)
             => _jsComponentInterop.AddRootComponent(identifier, domElementSelector);
 
         [JSInvokable] // Linker preserves this if you call RootComponents.Add
-        public void SetRootComponentParameters(int componentId, int parameterCount, JsonElement parametersJson)
+        public void SetJSRootComponentParameters(int componentId, int parameterCount, JsonElement parametersJson)
             => _jsComponentInterop.SetRootComponentParameters(componentId, parameterCount, parametersJson, _jsonOptions);
 
         [JSInvokable] // Linker preserves this if you call RootComponents.Add
-        public void RemoveRootComponent(int componentId)
+        public void RemoveJSRootComponent(int componentId)
```

Comment:
These changes aren't final, like many of the current changes on the .NET side. The C# changes will be more ready-to-review when I finish WebAssembly support for this feature.